packages/generators/app/lib/utils/git.js

```diff
@@ -0,0 +1,59 @@
+const { execSync } = require('child_process');
+'use strict';
+
+const path = require('path');
+const rimraf = require('rimraf');
+
+function isInGitRepository() {
+  try {
+    execSync('git rev-parse --is-inside-work-tree', { stdio: 'ignore' });
+    return true;
+  } catch (_) {
+    return false;
+  }
+  return false;
+}
+
+function isInMercurialRepository() {
+  try {
+    execSync('hg --cwd . root', { stdio: 'ignore' });
+    return true;
+  } catch (_) {
+    return false;
+  }
+  return false;
+}
+
+function tryGitInit(root) {
+  let didInit = false;
+  try {
+    execSync('git --version', { stdio: 'ignore' });
+    if (isInGitRepository() || isInMercurialRepository()) {
+      return false;
+    }
+
+    execSync('git init', { stdio: 'ignore' });
+    didInit = true;
+
+    execSync('git checkout -b main', { stdio: 'ignore' });
+
+    execSync('git add -A', { stdio: 'ignore' });
+    execSync('git commit -m "Initial commit from Create Strapi App"', {
```

Comment:
We shouldn't commit for them, many companies have very strict standards on commit history messages.

Reply:
I think we should not prevent 99% of our users from benefiting from a better experience because of a constraint of 1% of our users (who can still remove the first commit if they are not comfortable with it).

Reply:
I don't see the major benefit of doing a git commit for somebody on a creation no other project does that.

I get initializing the get repo itself but to add a bunch of files and do a commit for them just seems like overstepping. These things take like 3 seconds to do normally from the user side.

Reply:
They indeed take 3 seconds, but they are still 3 seconds we can save for every user. Also, my guess is that many users do not understand Strapi is designed to be versioned through Git and then deployed in production. This will hopefully help a bit. Projects like Next.js init the Git repository and trigger the initial commit, which improves the developer experience in my opinion. For Strapi, doing the initial commit will allow users to see the diff once they create their first content type(s).

Reply:
I removes that part for now because Strapi generates files after the app is created and this is quite weird to make a commit and right after the app (ts type generation for ex) the user needs to do a new one. Let's start small and see if we get comment about that.